### ISSUE TITLE
a11y/CTA titles & alt Ttext on Figure component

### ISSUE THREAD

**Issue by zlonko** [sourcegraph/about#5554]
What?
- Titles need to be added to items missing them. Source: [CTA audit spreadsheet](https://docs.google.com/spreadsheets/d/1N4Bgt74f1XDV3B3bnn5rNT5XoV_2Jfl8Rfk5-WTYG38/edit#gid=0). 
- We will also update the `Figure` component to surface a title on a link. This will ensure titles are captured by a screen reader.

Why? 
- Titles on CTAs and images must be picked up by a screen reader per a11y standards.
